src/legacy/core_plugins/kibana/public/dashboard/np_ready/dashboard_state_manager.ts

```diff
@@ -175,6 +175,14 @@ export class DashboardStateManager {
           // sync state required state container to be able to handle null
           // overriding set() so it could handle null coming from url
           if (state) {
+            // Skip this update if current dashboardId in the url is different from what we have in the current instance of state manager
+            // As dashboard is driven by angular at the moment, the destroy cycle happens async,
+            // If the dashboardId has changed it means this instance
+            // is going to be destroy soon and we shouldn't sync state anymore,
+            // as it could potentially trigger further url updates
+            const currentDashboardIdInUrl = getDashboardIdFromUrl(history.location.pathname);
```

Comment:
Yes, this is very similar to this fix we did before: https://github.com/elastic/kibana/pull/57795
Just covers another edge case.

I really hope this will go away with full dashboard migration to react